### PR TITLE
rosbag2_py: parametrize tests across storage plugins

### LINKED ISSUE #1185: Add integration test for mcap storage plugin
The latest release of mcap storage plugin is currently broken on one or more distributions (https://github.com/ros-tooling/rosbag2_storage_mcap/issues/82).

We need to add an integration test to CI & the buildfarm that will prevent such issues in the future.

### PR DESCRIPTION
**Pull request by james-rms**
Signed-off-by: James Smith <james@foxglove.dev>

Fixes #1185

### CODE REVIEW

**Comment by james-rms** [comment]
Gist: https://gist.githubusercontent.com/james-rms/f45f8da36160bd2f8b9b3f2a359436fa/raw/a67fe84db1a89be14af7d45931b1d7ff82a24fa4/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_test_common rosbag2_py ros2bag
TEST args:  --packages-above rosbag2_test_common rosbag2_py ros2bag
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/11256
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17800/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12363/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18456/)

File: rosbag2_py/test/test_convert.py [hunk 1]
```diff
@@ -65,29 +63,29 @@ def test_output_bags_not_a_list(self):
...
-  exclude: ".*empty.*"
-"""
-        with output_options_path.open('w') as f:
-            f.write(output_options_content)
-        bag_rewrite(input_options, str(output_options_path))
-        self.assertTrue(output_uri_1.exists() and output_uri_1.is_dir())
-        self.assertTrue((output_uri_1 / 'metadata.yaml').exists())
-        self.assertTrue(output_uri_2.exists() and output_uri_2.is_dir())
-        self.assertTrue((output_uri_2 / 'metadata.yaml').exists())
+        for storage_id in TESTED_STORAGE_IDS:
```
Comment (emersonknapp) [inline rosbag2_py/test/test_convert.py:66]: Is there a reason this one can't be parametrized?

**Review by emersonknapp — APPROVED** [review]
LGTM with green CI. Maybe one test could also be parametrized?

**Comment by james-rms** [comment]
Gist: https://gist.githubusercontent.com/james-rms/d27fa514c06f467e62698a498ed0bced/raw/a67fe84db1a89be14af7d45931b1d7ff82a24fa4/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_test_common rosbag2_py ros2bag
TEST args:  --packages-above rosbag2_test_common rosbag2_py ros2bag
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/11266
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17811/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12377/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18470/)

**Comment by james-rms** [comment]
Gist: https://gist.githubusercontent.com/james-rms/de95c8fbfcfc86b9cf90e41af5669ba2/raw/a67fe84db1a89be14af7d45931b1d7ff82a24fa4/ros2.repos
BUILD args:  --packages-above-and-dependencies rosbag2_py ros2bag
TEST args:  --packages-above rosbag2_py ros2bag
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/11268
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/17813/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12379/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18472/)

**Pull request merged**